### ISSUE TITLE
Decoder channels

### ISSUE THREAD

**Issue by AvivSham** [xiaoyufenfei/LEDNet#7]
I tried to implement LEDNet in keras and found that I have ~2.5m parameters which is incorrect according to the paper (claim to have ~900K). I saw you wrote two versions of LEDNet one with ~900K and the other like my implementation with ~2.5m. In the 900K implementation, you downsampled the number of channels to 1 straight away from the first conv layer in the downsampling branch of the decoder (why is that?). In the attached figure (from the paper) the number of the channels should remain the same through the first two levels of the downsampling branch of the decoder (and they still claim to have just ~900k parameters). What am I missing here? is this problematic paper? 

![image](https://user-images.githubusercontent.com/43371254/59549629-02a1b180-8f69-11e9-86be-691458668ced.png)

**Comment by wpf535236337** [comment]
@AvivSham i agree with you. the decoder network in Fig.1 has a problem，the channels is 1 not 128. otherwise, the  total parameter is about 2.5M. as far as I understood, the middle branch ‘128x64xC’ retains the most information, so the left branch may just learns a spatial-wise multiplier.

**Comment by AvivSham** [comment]
any comments?

**Comment by wpf535236337** [comment]
@AvivSham  i think you are right，two version：0.9M->lednet.py, 2.5M->lednet_1.py. the paper may have a problem while the author can‘t offer the pretrained model which has the right parameters and claimed accuracy。

**Comment by AvivSham** [comment]
@wpf535236337  in that case it's a really amateur work from the author from my point of view. you cannot claim something and present figure which tells other thing. The whole point of this paper is a low number of parameters 2.5M is not that lightweight.

**Comment by buzhiqimeiliuqiangdong** [comment]
> @wpf535236337 in that case it's a really amateur work from the author from my point of view. you cannot claim something and present figure which tells other thing. The whole point of this paper is a low number of parameters 2.5M is not that lightweight.

I tried to take the APN module separately and then did an experiment to calculate the parameter quantity. It turns out that the parameter amount has reached more than 1.3M in the case of the APN module, not to mention the previous layers.

**Comment by AvivSham** [comment]
@buzhiqimeiliuqiangdong  what is the total amount of parameters in your model? if it's 2.5M it is totally fine and if you want to reach to 900K parameters model some modifications needed.
I can help you with that in private.

**Comment by buzhiqimeiliuqiangdong** [comment]
@AvivShamThank you very much, I didn't reproduce his code, I just want to use the APN in his model, but this idea needs to be shelved, I have been busy with other experiments recently, thank you again.

**Comment by AvivSham** [comment]
NP glad to help :)
